### ISSUE TITLE
Bibliography not displayed if no in-text citations

### ISSUE THREAD

**Issue by adomasven** [zotero/zotero#1213]
Moved from zotero/zotero-word-for-windows-integration#37

### Problem No. 1:

1. Create a fresh document
1. Insert a citation
1. Insert bibliography
1. Add more entries to bibliography via Edit Bibliography interface
1. Remove the citation
1. Refresh

**Result**:
In place of bibliography we display {Bibliography}. If you remove the {Bibliography} field and click "Insert Bibliography" again {Bibliography} is inserted again.

**Proposed change**:
Custom entries are displayed

---

### Problem No. 2

1. Create a fresh document
1. Attempt to insert bibliography

**Result**:
Error displayed informing you need to insert citations first

**Proposed change**:
{Bibliography} is inserted, which can then be edited to add custom entries

---

There have been [reports](https://forums.zotero.org/discussion/comment/272903/#Comment_272903) of this on the forums recently.

Comments welcome.

**Comment by dstillman** [comment]
> Proposed change:
>{Bibliography} is inserted, which can then be edited to add custom entries

I don't think we should ever display `{Bibliography}` to the user (except in transition where we can't help it). If we're going to fix 2, a better solution would be to just pop up the add/edit citation dialog immediately. If they add a citation, great. If they cancel, nothing is inserted into the document. If they edit an existing bibliography and remove all custom entries and there are no in-text citations…maybe we could display `{Bibliography}` at that point. (Or we could just delete the bibliography, but `{Bibliography}` seems somewhat reasonable there.)

But the main danger of this is people thinking this is how they should be creating a bibliography, instead of creating in-text citations. I think there's actually a significant risk of this creating more confusion. We should obviously fix the ways that this results in brokenness, but do we have reason to think people need to create bibliographies this way?

/cc @adam3smith

**Comment by adomasven** [comment]
> But the main danger of this is people thinking this is how they should be creating a bibliography, instead of creating in-text citations. I think there's actually a significant risk of this creating more confusion.

That's a good point.

> If we're going to fix 2, a better solution would be to just pop up the add/edit citation dialog immediately.

2 isn't a problem itself, but rather a point of inconsistency. Once you've inserted a citation, you can (re)insert the bibliography as much as you want, even if you deleted all in-text citations (at which point it just inserts {Bibliography}).

**The alternative proposed solution is:**
When all citations are removed, remove the Bibliography field. Upon attempted inserts prompt that a citation needs to be inserted.

**Comment by dstillman** [comment]
That seems fine to me.

**Comment by adomasven** [comment]
(1) now throws an `Integration.Session.editBibliography: called without loaded citations` error instead.

See https://forums.zotero.org/discussion/74445/libreoffice-extension-bug-integration-session-editbibliography-called-without-loaded-citations
